lib/CL/pocl_networking.c

```diff
@@ -124,7 +218,31 @@ pocl_remote_client_set_socket_options (int socket_fd, int bufsize, int is_fast)
       (setsockopt (socket_fd, SOL_SOCKET, SO_KEEPALIVE, &one, sizeof (one))),
       -1, "setsockopt(SO_KEEPALIVE) returned errno: %i\n", errno);
 
-  unsigned int user_timeout = 10 * SECONDS_TO_MS;
+  POCL_RETURN_ERROR_ON ((setsockopt (socket_fd, SOL_SOCKET, SO_RCVTIMEO,
+                                     (const char *)&tv, sizeof (tv))),
+                        -1, "setsockopt(SO_RCVTIMEO) returned errno: %i\n",
+                        errno);
+  POCL_RETURN_ERROR_ON ((setsockopt (socket_fd, SOL_SOCKET, SO_SNDTIMEO,
+                                     (const char *)&tv, sizeof (tv))),
+                        -1, "setsockopt(SO_SNDTIMEO) returned errno: %i\n",
+                        errno);
+
+#ifdef ENABLE_VSOCK
+  if (ai_family != AF_VSOCK)
+    {
```

Comment:
It'd be cleaner to only `#ifdef` the if statement here (and its closing brace further down) rather than all the code in it, since the current solution causes unnecessary duplication of the TCP socket options.

Maybe even just do an early return if this is a vsock socket for now?